src/ResourceQuantity.cs

```diff
@@ -175,8 +178,16 @@ public string CanonicalizeString(SuffixFormat suffixFormat)
 
         // ctor
         partial void CustomInit()
-        {
-            var value = (Value ?? "").Trim();
+        {
+            if (Value == null)
```

Comment:
`new ResourceQuantity()` and `''` are not valid in previous design 
so, you want to make `new ResourceQuantity() === 0`?

Reply:
`new ResourceQuantity()` would throw `System.FormatException: 'The string '' cannot be converted to fraction.'` in the previous version. I think that's a very confusing thing to happen, as the constructor doesn't take any parameter.

Either you allow for a parameter-less constructor, which doesn't throw, or you make the constructor private, so the parameter is not callable.

Calling `new ResourceQuantity('')` will throw the same exception both before and with the changes in this PR. I've added a unit test to make that behavior explicit.

That said, the way the YAML deserialize seems to work is that it creates a new instance of your object using the default constructor and then calls `Read` on that instance. That won't work if the default constructor throws, so that's why I made this change.

Reply:
agree.